2-mandatory/6-journey-planner.js

```diff
@@ -122,7 +128,10 @@ function getLocationName() {}
   Advanced challange: try to use arrow function when invoking an array method.
 */
 function journeyPlanner(locations, transportMode) {
-  // Implement the function body
+  let transportArray = locations
+    .filter((x) => getTransportModes(x).some((y) => y === transportMode))
+    .map((x) => x[0]);
+  return transportArray;
```

Comment:
Good work - this is a complex problem and you have clearly understood how to use and chain the different methods.